### PR TITLE
docs(datepicker): add demo for onHidden/onShown outputs

### PR DESCRIPTION
**Pull request by ghost**
add demo for onHidden/onShown outputs related to #3607

### CODE REVIEW

File: demo/src/app/components/+datepicker/demos/onHidden-onShown/hidden-shown.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-datepicker-hidden-shown',
+  templateUrl: './hidden-shown.component.html'
+})
+export class DemoDatePickerHiddenShownComponent{
+  state = 'close';
+  onShowDateDatePicker(): void {
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/demos/onHidden-onShown/hidden-shown.component.ts]: just `onShown()` and `onHidden()`

File: demo/src/app/components/+datepicker/demos/index.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ import { DemoDatepickerDisabledComponent } from './disabled/disabled.component';
 import { DemoDatepickerFormsComponent } from './forms/forms.component';
 import { DemoDatepickerReactiveFormsComponent } from './reactive-forms/reactive-forms.component';
 import { DemoDatePickerCustomFormatComponent } from './custom-format/date-picker-custom-format';
+import { DemoDatePickerHiddenShownComponent } from './onHidden-onShown/hidden-shown.component';
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/demos/index.ts]: only lowercase letters

File: demo/src/app/components/+datepicker/datepicker-section.list.ts [hunk 3]
```diff
@@ -131,7 +132,15 @@ export const demoComponentContent: ContentSection[] = [
         component: require('!!raw-loader?lang=typescript!./demos/reactive-forms/reactive-forms.component.ts'),
         html: require('!!raw-loader?lang=markup!./demos/reactive-forms/reactive-forms.component.html'),
         outlet: DemoDatepickerReactiveFormsComponent
-      }
+      },
+      {
+        title: 'onHidden/onShown',
+        anchor: 'hidden-shown',
+        component: require('!!raw-loader?lang=typescript!./demos/onHidden-onShown/hidden-shown.component.ts'),
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/datepicker-section.list.ts]: Ts and html files in demo folder must have the same names

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by EvilAlexei — CHANGES_REQUESTED** [review]
Typo in heading `Shon`Hidden

**Review by EvilAlexei — CHANGES_REQUESTED** [review]
I think visibility-events will be better than hidden-shown

File: demo/src/app/components/+datepicker/datepicker-section.list.ts [hunk 4]
```diff
@@ -131,7 +132,15 @@ export const demoComponentContent: ContentSection[] = [
...
         html: require('!!raw-loader?lang=markup!./demos/reactive-forms/reactive-forms.component.html'),
         outlet: DemoDatepickerReactiveFormsComponent
-      }
+      },
+      {
+        title: 'Visibility Events',
+        anchor: 'visibility-events',
+        component: require('!!raw-loader?lang=typescript!./demos/visibility-events/visibility-events.ts'),
+        html: require('!!raw-loader?lang=markup!./demos/visibility-events/visibility-events.html'),
+        description: `<p>You can add events when a datepicker show or hide</p>`,
```
Comment (IlyaSurmay) [inline demo/src/app/components/+datepicker/datepicker-section.list.ts]: Change description to `You can subscribe to datepicker's visibility events`

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Review by EvilAlexei — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Please, make output for events more clearer, not looks like another one title (use for it another font or background)
howitlooksnow
Two examples how it better to be:
oneexamplemoreclear
mostunderstandable

**Review by IlyaSurmay — DISMISSED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Needs to be updated from latest dev and fixed 
b2354188a4
- when I did it locally I got file conflicts:

**Comment by ghost** [comment]
update and testing my branch according to Yevheniia comments(conflicts in files) which I can't see

**Review by IlyaSurmay — DISMISSED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Review by valorkin — APPROVED** [review]

**Pull request merged**